internal/ingress/nats/nats_test.go

```diff
@@ -0,0 +1,59 @@
+package nats
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kelindar/talaria/internal/config"
+	"github.com/kelindar/talaria/internal/monitor"
+	"github.com/nats-io/nats-server/v2/server"
+	natsserver "github.com/nats-io/nats-server/v2/test"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/assert"
+)
+
+const TEST_PORT = 8369
+
+func RunServerOnPort(port int) *server.Server {
+	opts := natsserver.DefaultTestOptions
+	opts.Port = port
+	opts.JetStream = true
+	return RunServerWithOptions(&opts)
+}
+
+func RunServerWithOptions(opts *server.Options) *server.Server {
+	return natsserver.RunServer(opts)
+}
+
+func TestSubscribe(t *testing.T) {
+	s := RunServerOnPort(TEST_PORT)
+	defer s.Shutdown()
+
+	sUrl := fmt.Sprintf("nats://127.0.0.1:%d", TEST_PORT)
+
+	conf := &config.NATS{
```

Comment:
these can be set using config.Load and os.env stuff.

Reply:
This is just a mock nats server which only used in unit test of nats.

Reply:
Okayy, noted.